polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/PageTokenUtil.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.persistence.pagination;
+
+import static com.google.api.client.util.Preconditions.checkArgument;
+
+import jakarta.annotation.Nullable;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+
+final class PageTokenUtil {
+
+  private PageTokenUtil() {}
+
+  static PageToken decodePageRequest(
+      @Nullable String requestedPageToken, @Nullable Integer requestedPageSize) {
+    int pageSize;
+    String encodedDataReference = null;
```

Comment:
If a token has multiple components, should that be modeled as one "encoded data reference"? Multiple?

Reply:
Since we're defining the serde/deserde of the page token now, let's do it in a way that captures future page tokens we think are reasonable. I think that includes page requests with more than 2 discrete pieces of information.

Reply:
The current approach does allow for future extensions. Different "data pointers" can be introduced when a Persistence impl. needs them. Since the "data pointer" always makes a round trip from Persistence code to the client and back to the _same_ persistence code, that code is free to introduce new encoding/decoding methods without affecting anything else.

It is remotely possible that a page token ends up in an API method call different from the one that produced it. However, having "data pointer" sub-types (as opposed to just pairs of encode/decode methods) does not add any safety here. We could encode some "place" identifier into the token to bind it firmly to the API method, but I think it's an overkill.